src/main/java/com/redhat/foreman/ForemanCleanupThread.java

```diff
@@ -82,4 +92,201 @@ public void run() {
         Futures.getUnchecked(Futures.successfulAsList(nodeTerminationBuilder.build()));
     }
 
+//    @Initializer(after = InitMilestone.COMPLETED)
+//    public static void onCompleted() {
+//        LOGGER.finer("[START] ForemanCleanupThread.onCompleted()");
+//
+//        // Make the time consuming operation in the separate thread to not block other listeners
+//        new Thread("ForemanStartCleanupThread") {
+//            @Override
+//            public void run() {
+//                runCleanup();
+//            }
+//        }.start();
+//        LOGGER.finer("[COMPLETED] ForemanCleanupThread.onCompleted()");
+//    }
+
+    // Until resolved JENKINS-37759, then remove this class and use above onComplete()
+    @Extension
+    public final static class OnLoadedListener extends ItemListener {
+        static boolean executed = false;
+
+        @Override
+        synchronized public void onLoaded() {
```

Comment:
I presume you synchronize to protect `executed` and that is used to ensure this is run only only once. If I am not mistaken this will cause that firs job that gets here will block all other from loading. It is supposed to execute quickly, I know, but is still looks nasty. Since initializer seems broken, how about `Plugin#start`?